src/libs/SearchQueryUtils.ts

```diff
@@ -340,6 +341,39 @@ function isFilterSupported(filter: SearchAdvancedFiltersKey, type: SearchDataTyp
     return ALLOWED_TYPE_FILTERS[type].some((supportedFilter) => supportedFilter === filter);
 }
 
+/**
+ * Normalizes the groupBy value into a single string.
+ * - If groupBy is a string, it is returned as is.
+ * - If groupBy is an array, only the first element is taken and validated against CONST.SEARCH.GROUP_BY.
+ * - If no valid value is found, returns an undefined.
+ *
+ * This ensures consistent usage of groupBy across the app,
+ * since we only support filtering by a single valid groupBy key.
+ *
+ * @param groupBy - The raw groupBy value from SearchQueryJSON
+ * @returns The normalized groupBy value or undefined
+ */
+function getGroupByValue(groupBy?: string | string[]): SearchGroupBy | undefined {
+    if (!groupBy) {
+        return undefined;
```

Comment:
```suggestion
        return;
```

No need to explicitly return `undefined`.